apps/rush-lib/src/scripts/install-run.ts

```diff
@@ -63,45 +62,57 @@ function _parsePackageSpecifier(rawPackageSpecifier: string): IPackageSpecifier
+      for (const token of environmentVariables) {
+        // Remove the leading "${" and the trailing "}" from the token
+        const environmentVariableName: string = token.substring(2, token.length - 1);
+        if (!process.env[environmentVariableName]) {
+          lineShouldBeTrimmed = true;
+          break;
+        }
+      }
+    }
+
+    if (lineShouldBeTrimmed) {
+      // Example output:
+      // "; MISSING ENVIRONMENT VARIABLE: //my-registry.com/npm/:_authToken=${MY_AUTH_TOKEN}"
+      resultLines.push('; MISSING ENVIRONMENT VARIABLE: ' + line);
+    } else {
+      resultLines.push(line);
+    }
+  }
+
+  fs.writeFileSync(targetNpmrcPath, resultLines.join(os.EOL));
+}
+
+/**
+ * syncNpmrc() copies the .npmrc file to the target folder, and also trims unusable lines from the .npmrc file.
+ * If the source .npmrc file not exist, then syncNpmrc() will delete an .npmrc that is found in the target folder.
+ *
+ * IMPORTANT: THIS CODE SHOULD BE KEPT UP TO DATE WITH _syncNpmrc() FROM scripts/install-run.ts
  */
 function _syncNpmrc(sourceNpmrcFolder: string, targetNpmrcFolder: string): void {
   const sourceNpmrcPath: string = path.join(sourceNpmrcFolder, '.npmrc');
   const targetNpmrcPath: string = path.join(targetNpmrcFolder, '.npmrc');
   try {
     if (fs.existsSync(sourceNpmrcPath)) {
-      let npmrcFileLines: string[] = fs.readFileSync(sourceNpmrcPath).toString().split('\n');
-      npmrcFileLines = npmrcFileLines.map((line) => (line || '').trim());
-      const resultLines: string[] = [];
-      // Trim out lines that reference environment variables that aren't defined
-      for (const line of npmrcFileLines) {
-        // This finds environment variable tokens that look like "${VAR_NAME}"
-        const regex: RegExp = /\$\{([^\}]+)\}/g;
-        const environmentVariables: string[] | null = line.match(regex);
-        let lineShouldBeTrimmed: boolean = false;
-        if (environmentVariables) {
-          for (const token of environmentVariables) {
-            // Remove the leading "${" and the trailing "}" from the token
-            const environmentVariableName: string = token.substring(2, token.length - 1);
-            if (!process.env[environmentVariableName]) {
-              lineShouldBeTrimmed = true;
-              break;
-            }
-          }
-        }
-
-        if (lineShouldBeTrimmed) {
-          // Example output:
-          // "; MISSING ENVIRONMENT VARIABLE: //my-registry.com/npm/:_authToken=${MY_AUTH_TOKEN}"
-          resultLines.push('; MISSING ENVIRONMENT VARIABLE: ' + line);
-        } else {
-          resultLines.push(line);
-        }
-      }
-
-      fs.writeFileSync(targetNpmrcPath, resultLines.join(os.EOL));
+      _copyAndTrimNpmrcFile(sourceNpmrcPath, targetNpmrcPath);
     } else if (fs.existsSync(targetNpmrcPath)) {
       // If the source .npmrc doesn't exist and there is one in the target, delete the one in the target
+      console.log(`Deleting ${targetNpmrcPath}`);
```

Comment:
Is this log message useful?

Reply:
The earlier  `_syncNpmrc()` in install-run script did not have log messages. Is there a specific reason why (maybe for CI purposes)?

Reply:
I guess they aren't doing any harm. Can you add a `// Verbose` comment to these, so when we replace the logger in Rush with `Terminal`, we'll know to redirect these to verbose?

```suggestion
      console.log(`Deleting ${targetNpmrcPath}`); // Verbose
```

Reply:
Done.